[Bug?]: Importing micromatch in third party plugin breaks the builder

### Self-service

- [X] I'd be willing to implement a fix

### Describe the bug

When importing micromatch in a third party plugin, the `@yarnpkg/builder` `build plugin` command fails with the following error:

```
➤ YN0000: ┌ Building @yarnpkg/plugin-helloworld
➤ YN0001: │ Error: Build failed with 5 errors:
node_modules/fill-range/index.js:10:21: error: Could not resolve "util" (use "platform: 'node'" when building for node)
node_modules/micromatch/index.js:3:21: error: Could not resolve "util" (use "platform: 'node'" when building for node)
node_modules/picomatch/lib/constants.js:3:21: error: Could not resolve "path" (use "platform: 'node'" when building for node)
node_modules/picomatch/lib/picomatch.js:3:21: error: Could not resolve "path" (use "platform: 'node'" when building for node)
node_modules/picomatch/lib/utils.js:3:21: error: Could not resolve "path" (use "platform: 'node'" when building for node)
    at failureErrorWithLog (/Users/mark/dev/plugin-build-test/node_modules/esbuild-wasm/lib/main.js:1443:15)
    at /Users/mark/dev/plugin-build-test/node_modules/esbuild-wasm/lib/main.js:1125:28
    at runOnEndCallbacks (/Users/mark/dev/plugin-build-test/node_modules/esbuild-wasm/lib/main.js:915:63)
    at buildResponseToResult (/Users/mark/dev/plugin-build-test/node_modules/esbuild-wasm/lib/main.js:1123:7)
    at /Users/mark/dev/plugin-build-test/node_modules/esbuild-wasm/lib/main.js:1230:14
    at /Users/mark/dev/plugin-build-test/node_modules/esbuild-wasm/lib/main.js:606:9
    at handleIncomingPacket (/Users/mark/dev/plugin-build-test/node_modules/esbuild-wasm/lib/main.js:703:9)
    at Socket.readFromStdout (/Users/mark/dev/plugin-build-test/node_modules/esbuild-wasm/lib/main.js:573:7)
    at Socket.emit (events.js:315:20)
    at addChunk (internal/streams/readable.js:309:12)
➤ YN0000: └ Completed in 0s 704ms

➤ YN0001: ✘ Failed to build @yarnpkg/plugin-helloworld
```

### To reproduce

A reproduction repo can be found here: https://github.com/mskelton/plugin-build-test.

### Environment

  System:
    OS: macOS 12.0
    CPU: (8) x64 Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
  Binaries:
    Node: 14.16.0 - /private/var/folders/cj/22br0t1d2xlcqyncvdm5q3mc0000gn/T/xfs-e7a6d128/node
    Yarn: 3.0.0 - /private/var/folders/cj/22br0t1d2xlcqyncvdm5q3mc0000gn/T/xfs-e7a6d128/yarn
    npm: 7.20.0 - /usr/local/bin/npm
  npmPackages:
    jest: ^27.0.6 => 27.0.6 

### Additional context

_No response_

--- a/.yarn/versions/9fef648d.yml
+++ b/.yarn/versions/9fef648d.yml
@@ -1,0 +1,13 @@
+releases:
+  "@yarnpkg/builder": patch
+
+declined:
+  - "@yarnpkg/plugin-constraints"
+  - "@yarnpkg/plugin-exec"
+  - "@yarnpkg/plugin-interactive-tools"
+  - "@yarnpkg/plugin-stage"
+  - "@yarnpkg/plugin-typescript"
+  - "@yarnpkg/plugin-version"
+  - "@yarnpkg/plugin-workspace-tools"
+  - "@yarnpkg/cli"
+  - "@yarnpkg/pnp"

--- a/packages/yarnpkg-builder/sources/commands/build/bundle.ts
+++ b/packages/yarnpkg-builder/sources/commands/build/bundle.ts
@@ -121,6 +121,7 @@ export default class BuildBundleCommand extends Command {
           define: {YARN_VERSION: JSON.stringify(version)},
           outfile: output,
           logLevel: `silent`,
+          format: `iife`,
           platform: `node`,
           plugins: [valLoader, pnpPlugin()],
           minify: !this.noMinify,

--- a/packages/yarnpkg-builder/sources/commands/build/bundle.ts
+++ b/packages/yarnpkg-builder/sources/commands/build/bundle.ts
@@ -121,6 +121,7 @@ export default class BuildBundleCommand extends Command {
           define: {YARN_VERSION: JSON.stringify(version)},
           outfile: output,
           logLevel: `silent`,
+          platform: 'node',
           plugins: [valLoader, pnpPlugin()],
           minify: !this.noMinify,
           sourcemap: this.sourceMap ? `inline` : false,

--- a/packages/yarnpkg-builder/sources/commands/build/bundle.ts
+++ b/packages/yarnpkg-builder/sources/commands/build/bundle.ts
@@ -121,7 +121,7 @@ export default class BuildBundleCommand extends Command {
           define: {YARN_VERSION: JSON.stringify(version)},
           outfile: output,
           logLevel: `silent`,
-          platform: 'node',
+          platform: `node`,
           plugins: [valLoader, pnpPlugin()],
           minify: !this.noMinify,
           sourcemap: this.sourceMap ? `inline` : false,

--- a/packages/yarnpkg-builder/sources/commands/build/plugin.ts
+++ b/packages/yarnpkg-builder/sources/commands/build/plugin.ts
@@ -113,6 +113,7 @@ export default class BuildPluginCommand extends Command {
           bundle: true,
           outfile: output,
           logLevel: `silent`,
+          format: `iife`,
           platform: `node`,
           plugins: [dynamicLibResolver, pnpPlugin()],
           minify: !this.noMinify,

--- a/packages/yarnpkg-builder/sources/commands/build/plugin.ts
+++ b/packages/yarnpkg-builder/sources/commands/build/plugin.ts
@@ -113,7 +113,7 @@ export default class BuildPluginCommand extends Command {
           bundle: true,
           outfile: output,
           logLevel: `silent`,
-          platform: 'node',
+          platform: `node`,
           plugins: [dynamicLibResolver, pnpPlugin()],
           minify: !this.noMinify,
           sourcemap: this.sourceMap ? `inline` : false,

--- a/packages/yarnpkg-builder/sources/commands/build/plugin.ts
+++ b/packages/yarnpkg-builder/sources/commands/build/plugin.ts
@@ -113,6 +113,7 @@ export default class BuildPluginCommand extends Command {
           bundle: true,
           outfile: output,
           logLevel: `silent`,
+          platform: 'node',
           plugins: [dynamicLibResolver, pnpPlugin()],
           minify: !this.noMinify,
           sourcemap: this.sourceMap ? `inline` : false,

